group-coordinator/src/main/java/org/apache/kafka/coordinator/group/metrics/GroupCoordinatorRuntimeMetrics.java

```diff
@@ -208,8 +209,8 @@ public void recordEventQueueTime(long durationMs) { }
     public void recordEventQueueProcessingTime(long durationMs) { }
 
     @Override
-    public void recordThreadIdleRatio(double ratio) {
-        threadIdleRatioSensor.record(ratio);
+    public synchronized void recordThreadIdleTime(long idleTimeMs, long currentTimeMs) {
```

Comment:
Why do we need to synchronize here?

Reply:
See the other comment. It is because we removed the sensor